### DISCUSSION TITLE
Add OpenEXR Support

### DISCUSSION THREAD

**Discussion by NHodgesVFX** [SixLabors/ImageSharp#2009] [Ideas]
OpenEXR is a format used heavily in VFX, animation, film, and game pipelines. Adding support for OpenEXR would allow ImageSharp to be used in those pipelines. Additionally it would add a second format for High Bit Depth exports, I think the only one currently able to export 32bpc is Tiff which has huge export sizes compared to exr.

Website:
https://www.openexr.com/

License:
Modified BSD

**Comment by JimBobSquarePants** [comment]
Cool format. This would have to be a community contribution though at this time.

**Comment by brianpopow** [comment]
I have started looking into support for OpenExr a while ago. I have a branch which could maybe a starting point for anyone interested to help in making this feature work for ImageSharp: https://github.com/SixLabors/ImageSharp/tree/bp/openExr

So far Decoding and Encoding of uncompressed OpenExr images work.
Also decoding zip and RLE compressed images work.

The real challenge would be the different other compressions. There is an official [Reference implementation](https://github.com/AcademySoftwareFoundation/openexr.git) which can help. 

For reference: [OpenExr Specification](https://github.com/SixLabors/ImageSharp/tree/bp/openExr)

**Comment by Raildex** [comment]
Any news on Exr support?

**Reply by brianpopow** [reply]
With #3096 merged, there is now initial support for decoding and encoding of EXR images.

Note that not all compression options are supported yet. For decoding the following compressions are supported:
- NO_COMPRESSION
- RLE_COMPRESSION
- ZIPS_COMPRESSION
- ZIP_COMPRESSION
- B44_COMPRESSION

Also some features are not yet supported, for example tiled images.

**Reply by JimBobSquarePants** [reply]
@brianpopow can you please open a tracking issue for that?

**Reply by brianpopow** [reply]
the tracking issue can be found [here](https://github.com/SixLabors/ImageSharp/issues/3116)
